client/state/data-layer/wpcom/account-recovery/lookup/index.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+
+import {
+	ACCOUNT_RECOVERY_RESET_OPTIONS_REQUEST,
+	ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE,
+	ACCOUNT_RECOVERY_RESET_OPTIONS_ERROR,
+} from 'state/action-types';
+
+const validate = ( { primary_email, primary_sms, secondary_email, secondary_sms } ) => [
+	primary_email,
+	primary_sms,
+	secondary_email,
+	secondary_sms,
+].every( String );
+
+export const fromApi = data => (
+	validate( data ) ? [
+		{
+			email: data.primary_email,
+			sms: data.primary_sms,
+			name: 'primary',
+		},
+		{
+			email: data.secondary_email,
+			sms: data.secondary_sms,
+			name: 'secondary',
+		},
+	] : [
+		{
+			email: '',
+			sms: '',
+			name: 'primary',
+		},
+		{
+			email: '',
+			sms: '',
+			name: 'secondary',
+		},
+	] );
```

Comment:
nice! I like the way you injected the validation step in here!

I wonder though about the error condition. if the data is invalid do we really want these empty values? (the answer could be yes, I just don't know)

also, you might check the linter as I don't think it will be happy with this ternary syntax alignment. I could imagine a cleaner way to write it would involve a composition of the independent validation and `fromApi()` mapping…

```js
const fromApi = data => isValid( data )
	? mapResponse( data )
	: emptyRecoveryInformation;
```

though seriously I would reconsider this. what is the point of injecting empty data into Redux? would we not already assume to have _no values_ in the absence of _real values_? what is the initial state of the reducers? if they store a string, should their initial state not already be the empty string?

a failure is a failure. we don't necessarily have to handle it, but we may not want to hide it either. _not_ dispatching is a valid way to follow-up on invalid data from the API, so is throwing an error